src/lib.rs

```diff
@@ -492,8 +489,8 @@ macro_rules! define_sentence_type_enum {
         }
 
         impl $Name {
-            fn try_from(s: &[u8]) -> Result<Self, &'static str> {
-                match str::from_utf8(s).map_err(|_| "invalid header")? {
+            fn try_from(s: &[u8]) -> Result<Self, NmeaError> {
+                match str::from_utf8(s).map_err(|_| NmeaError::Utf8DecodingError)? {
```

Comment:
Utf-8 error is strange for NMEA protocol, there is no mention of it.
May be just compare bytes instead? And remove `from_utf8` call at all, like:
```
    let x: &[u8] = ...;
    let res = match x {
        b"bbb" => false,
        b"aaaa" => true,
        b"ccc" => false,
        _ => false,
    };
```

Rust doesn't support compare bytes at the time this code was written,
but it supports now.

Reply:
Also reasonable, I shall get to this too!